src/interchange/src/avro.rs

```diff
@@ -1213,8 +1314,8 @@ struct DebeziumDeduplicationState {
 #[derive(Debug)]
 struct TrackFull {
     /// binlog filename to (offset to (timestamp that this binlog entry was first seen))
-    seen_offsets: HashMap<String, HashMap<(usize, usize), i64>>,
-    seen_snapshot_keys: HashMap<String, HashSet<Row>>,
+    seen_offsets: HashMap<Vec<u8>, HashMap<(usize, usize), i64>>,
+    seen_snapshot_keys: HashMap<Vec<u8>, HashSet<Row>>,
```

Comment:
Since we're modifying the types of this anyway it seems like maybe we should make this `Box<[u8]>` to more clear that these are immutable. It doesn't super matter since you can't get a mut reference to a hashmap key, but it is weird.

Reply:
I think you're right that that's more idiomatic